ftw/referencewidget/widget.py

```diff
@@ -78,19 +80,36 @@ def form_url(self):
         return self.form.request.getURL()
 
     def js_value(self):
+        storage = queryUtility(IRedirectionStorage)
         result = []
+
         if not self.value:
             return
+
         if isinstance(self.value, list):
             for item in self.value:
-                obj = self.context.unrestrictedTraverse(item.encode('utf8'))
-                result.append({'path': item.encode('utf8'),
-                               'title': obj.title})
+                path = item.encode('utf8')
+
+                obj = self.context.unrestrictedTraverse(path, None)
+
+                if obj is None:
+                    path = storage.get(path)
```

Comment:
@maethu since you are doing the exact same block of code twice it would be great to extract that to a separate method (e.g. `def get_object_by_path(self, path)`).